app/src/main/java/fr/free/nrw/commons/upload/worker/UploadWorker.kt

```diff
@@ -403,10 +420,33 @@ class UploadWorker(var appContext: Context, workerParams: WorkerParameters) :
                 }
                 else -> {
                     Timber.e("""upload file to stash failed with status: ${stashUploadResult.state}""")
-                    showFailedNotification(contribution)
+                    showInvalidLoginNotification(contribution)
                     contribution.state = Contribution.STATE_FAILED
                     contribution.chunkInfo = null
                     contributionDao.saveSynchronous(contribution)
+                    if (stashUploadResult.errorMessage.equals(appContext.getString(R.string.error_invalid_token))) {
```

Comment:
@shashankiitbhu Can you make the constant visible to this class, and use it instead of `R.string.error_invalid_token`? An internationalized string can change depending on the locale, and this match may fail because of that. (I believe @domdomegg's suggestion was to use resources for other strings, not this one.)